### ISSUE TITLE
Upgrade error: Realm SDK package version does not match the native library version

### ISSUE THREAD

**Issue by mnimer** [realm/realm-dart#1127]
### What happened?

Received this runtime error after trying to upgrade:
Realm SDK package version does not match the native library version (0.10.0+rc != 0.8.0+rc). This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues

Upgraded 0.8.0 to 0.10.0 following the steps on this page
https://www.mongodb.com/docs/realm/sdk/flutter/install/#update-package-version

I have already tried running:
- flutter clean 
- flutter pub cache repair
- flutter pub get

### Repro steps

Start with an existing project on ^0.8.0+rc and upgrade the project to 0.10.0, following these instructions: https://www.mongodb.com/docs/realm/sdk/flutter/install/#update-package-version

run flutter clean & build

### Version

^0.8.0+rc

### What Atlas Services are you using?

Local Database only

### What type of application is this?

Flutter Application

### Client OS and version

mac 13.1

### Code snippets

final config = Configuration.local(
          [Apps.schema, FileCollection.schema, Folder.schema, File.schema],
          schemaVersion: 1, shouldDeleteIfMigrationNeeded: true);
database = Realm(config);

### Relevant log output

```shell
Realm SDK package version does not match the native library version (0.10.0+rc != 0.8.0+rc). This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues
```

**Comment by nielsenko** [comment]
@mnimer Hi Mike

I have failed to recreate this issue. A simple `flutter pub upgrade --major-versions` and then running is enough for me.

Could you try to remove your platform projects, and then re-create them with:
```sh
flutter create . --platforms=android,ios,linux,macos,windows
```
or whatever platforms you support.

**Comment by mnimer** [comment]
'flutter create . --platforms=...' Was the trick! upgrade alone did not fix it but when I deleted the platform folders and rebuilt the macos platform it started working. Thank you.

**Comment by mnimer** [comment]
This broke again when I turned off macos sandbox mode
<key>com.apple.security.app-sandbox</key>
<false/>

Error: 
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: RealmException: Realm SDK package version does not match the native library version (0.11.0+rc != 0.8.0+rc). This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues
#0      new _RealmCore._ (package:realm/src/native/realm_core.dart:78:7)
#1      new _RealmCore (package:realm/src/native/realm_core.dart:83:37)
#2      realmCore (package:realm/src/native/realm_core.dart:52:30)
#3      realmCore (package:realm/src/native/realm_core.dart)
#4      Configuration.defaultStoragePath (package:realm/src/configuration.dart:100:14)
#5      Configuration.defaultRealmPath (package:realm/src/configuration.dart:110:47)
#6      Configuration.defaultRealmPath (package:realm/src/configuration.dart)
#7      Configuration.defaultRealmName= (package:realm/src/configuration.dart:85:91)

**Comment by nielsenko** [comment]
This is very surprising?! Does recreating the platform project still help?

BTW, you don't need to disable the security sandbox, as long as you add the entitlement to use the network, it will work:
```xml
...
<dict>
        ...
        <key>com.apple.security.network.client</key>
        <true/>
</dict>
...
```

**Comment by nielsenko** [comment]
If you run:
```sh
find ~/ -name librealm_dart.dylib
```
How many paths do you find, and if you feel comfortable doing so, could you share them here?

**Comment by mnimer** [comment]
Needed to turn off sandbox mode for another reason. Building a desktop file manager and couldn't access some images.  I did try to rebuild the platform and that did not fix it this time. 

Here are the paths:
/Users/mike//Development/github/fd-app/client/binary/macos/librealm_dart.dylib
/Users/mike//Development/github/fd-app/client/build/macos/Build/Products/Debug/client.app/Contents/Frameworks/realm.framework/Versions/A/Resources/librealm_dart.dylib
/Users/mike//Development/github/fd-app/client/build/macos/Build/Products/Debug/realm/realm.framework/Versions/A/Resources/librealm_dart.dylib
/Users/mike//.pub-cache/hosted/pub.dev/realm-0.8.0+rc/macos/librealm_dart.dylib
/Users/mike//.pub-cache/hosted/pub.dev/realm-0.10.0+rc/macos/librealm_dart.dylib
/Users/mike//.pub-cache/hosted/pub.dev/realm-0.11.0+rc/macos/librealm_dart.dylib

**Comment by nielsenko** [comment]
Could you try to remove 
```
/Users/mike//Development/github/fd-app/client/binary/macos/librealm_dart.dylib
/Users/mike//Development/github/fd-app/client/build
```
and report back?

**Comment by mnimer** [comment]
That seems to have done the trick, it is working now.

**Comment by koraybalci** [comment]
Same happened to me while trying to move from `1.5.0` to `1.6.0`. I had attempted to upgrade the package while I was testing in debug mode with android emulator. Hot restart did not help.

Here's what worked for me:
1. Stopped the debug mode (not the emulator entirely)
2. Ran `flutter clean`
3. Ran `flutter build apk` (not sure if it was relevant but I did observe `Downloading Realm binaries for realm@1.6.0 to C..` in that process)
4. Launched the app in emulator again in debug mode.
